sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -833,37 +841,36 @@ private void queueWork(int maximumMessageCount, Duration maxWaitTime,
 
         if (messageSubscriber != null) {
             messageSubscriber.queueWork(work);
-            return;
+            LOGGER.atVerbose().addKeyValue(WORK_ID_KEY, work.getId()).log("Receive request queued up.");
+            return messageSubscriber::processWorkQueue;
         }
 
-        messageSubscriber = synchronousMessageSubscriber.updateAndGet(subscriber -> {
-            // Ensuring we create SynchronousMessageSubscriber only once.
-            if (subscriber == null) {
-                return new SynchronousMessageSubscriber(asyncClient,
+        final boolean isFirstWork;
+        synchronized (createSubscriberLock) {
```

Comment:
Should we think about using a `Semaphore` rather than a synchronization block given their better support with virtual threads?

Reply:
https://docs.oracle.com/en/java/javase/21/core/virtual-threads.html#GUID-704A716D-0662-4BC7-8C7F-66EE74B1EDAD

Sorry, looks like `ReentrantLock` is the better option

Reply:
good idea, changed to `ReentrantLock`